app/controllers/posts_controller.rb

```diff
@@ -49,6 +49,14 @@ def create
     @post = Post.new(post_params.merge(user: current_user, body: helpers.post_markdown(:post, :body_markdown),
                                        category: @category, post_type: @post_type, parent: @parent))
 
+    if @post.title?
+      if @post.title.include? "$$"
+        flash[:danger] = helpers.i18ns('Title can not contain "$$"', type: @post_type.name)
```

Comment:
`helpers.i18ns` is a wrapper round `I18n.t` that substitutes text in before it's sent for translation. This call is trying to substitute `":type"` in the string for `@post_type.name`, but you haven't got and don't need `:type` in the string, so you can remove this entirely.

The argument to `i18ns` (and `I18n.t`) should be the _translation key_, not the string you want translated. The actual string needs to go in a locale file.

Let's also be a bit clearer about the error message.

```suggestion
        flash[:danger] = I18n.t 'posts.no_block_mathjax_title'
```

Then you also need to add another entry in `config/locales/strings/en.posts.yml`, like the following:

```yml
  no_block_mathjax_title: >
    Title cannot contain block-level MathJax.
```